tsdb/db.go

```diff
@@ -804,55 +804,57 @@ func (db *DB) Compact() (err error) {
 		// so in order to make sure that overlaps are evaluated
 		// consistently, we explicitly remove the last value
 		// from the block interval here.
-		head := NewRangeHead(db.head, mint, maxt-1)
-		if err := db.compactHead(head); err != nil {
-			return err
+		if err := db.compactHead(NewRangeHead(db.head, mint, lastBlockMaxt-1)); err != nil {
+			return errors.Wrap(err, "compact head")
```

Comment:
If there were multiple compactions happening on Head during to a long time range and consider the last one failed here, we would end up not truncating the WAL (hence the ugly defer earlier :/)

Reply:
Yea, true. Instead of fighting with weird vars what about putting this into function or closure `func() { }()` and defer here?